[BCDA-8824] Update GitHub actions

jhutchison-nava opened the pull request:
## 🎫 Ticket

https://jira.cms.gov/browse/BCDA-8824

## 🛠 Changes

Updates GitHub Actions to match AB2D deployment process.
- Adds a `deploy` script that stores `uswds-redesign` (for now until conversion to `main`) and PR's in relative S3 directories
- Adds a `promote` script that copies contents of `uswds-redesign` (for now until conversion to `main`) to production
- Updates the accessibility script

## ℹ️ Context

Simplifies and standardizes deployment process.

## 🧪 Validation

To be validated on the `uswds-redesign` branch during new site development

carlpartridge commented on .github/workflows/a11y-check.yml:
```diff
@@ -0,0 +1,68 @@
+            });
+
+            core.summary.addRaw(body).write();
+
```
Is this an artifact?  Do we want to save this as an actual file we can look at?  Or is just posting the contents to slack good enough?

jhutchison-nava replied:
These get saved to the Action summary. That just makes it easy to associate a specific run to specific results. I think that and posting to slack are good enough.

carlpartridge commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,73 @@
+      # Step 7: Deploy to Dev
+      - name: Copy site to Dev Bucket
+        run: |
+          aws s3 sync _site/ s3://stage.bcda.cms.gov/${{ inputs.deploy_path }}/ --delete
```
This deploy script only deploys to stage?  Do we want to change the description to "stage" (over "dev")?
Edit: Ok I see we promote it to prod in the promote workflow.

carlpartridge commented on .github/workflows/pull-request.yml:
```diff
@@ -0,0 +1,30 @@
+    secrets: inherit
+
+  # broken-link-check:
+  #   uses: ./.github/workflows/broken-link-check.yml
```
Any reason this is commented out?

jhutchison-nava replied:
I can add that back in. It wasn't set up in AB2D yet but since it was running on BCDA I can reenable it and copy it over

jhutchison-nava commented:
I updated the Deployment Process doc in confluence when I implemented this process on AB2D https://confluence.cms.gov/display/ODI/Deployment+Process

AB2D is slightly different that BCDA and CPD since AB2D has multiple AWS accounts for handling `dev` and `prod`, but I can make a note of that.

carlpartridge commented:
> I updated the Deployment Process doc in confluence when I implemented this process on AB2D https://confluence.cms.gov/display/ODI/Deployment+Process
> 
> AB2D is slightly different that BCDA and CPD since AB2D has multiple AWS accounts for handling `dev` and `prod`, but I can make a note of that.

FYI we have: https://confluence.cms.gov/pages/viewpage.action?spaceKey=BCDA&title=Technical+Process+for+Deploying+Releases
